lib/active_merchant/billing/gateways/cyber_source.rb

```diff
@@ -520,9 +520,11 @@ def add_purchase_data(xml, money = 0, include_grand_total = false, options = {})
       end
 
       def add_address(xml, payment_method, address, options, shipTo = false)
+        first_name, last_name = address_names(address[:name], payment_method)
+
         xml.tag! shipTo ? 'shipTo' : 'billTo' do
-          xml.tag! 'firstName',             payment_method.first_name if payment_method
-          xml.tag! 'lastName',              payment_method.last_name if payment_method
+          xml.tag! 'firstName',             first_name
+          xml.tag! 'lastName',              last_name
```

Comment:
Before, the XML `firstName` or `lastName` would not be set if not defined.
Does this set them even if not defined?

Reply:
You're right!